### PR TITLE
Add event handler and listener specs

### PR DESCRIPTION
**Pull request by sebastianfeldmann**

### CODE REVIEW

**Comment by sebastianfeldmann** [comment]
ping @icanhazstring

File: src/Hook/Event/Listener.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Interface EventListener
+ *
+ * @package CaptainHook
+ * @author  Sebastian Feldmann <sf@sebastian-feldmann.info>
+ * @link    https://github.com/captainhookphp/captainhook
+ * @since   Class available since Release 0.9.0
+ */
+interface Listener
```
Comment (icanhazstring) [inline src/Hook/Event/Listener.php:26]: This should be a `Subscriber` i think.
As it does not actually "listen" to events but does the subscription. Either on its own or pass it on to another handler.

**Review by icanhazstring — COMMENTED** [review]
Beside some naming bingo, looks nice to me 👍

**Comment by sebastianfeldmann** [comment]
@icanhazstring I "finished" it so far that it can be used I think.
I cleaned up the commit history so you can follow along the development process a bit better.

What do you think?

**Comment by icanhazstring** [comment]
Looks good to me. Like it 🙂

**Pull request merged**